dspace/config/spring/api/external-services.xml

```diff
@@ -52,12 +72,24 @@
         <property name="metadataSource" ref="pubmedImportService"/>
         <property name="sourceIdentifier" value="pubmed"/>
         <property name="recordIdMetadata" value="dc.identifier.other"/>
+        <property name="supportedEntityTypes">
+            <list>
+                <value>Publication</value>
+                <value>unset</value>
```

Comment:
@benbosman : Just to clarify what you mean by "null object", are you proposing that this line be changed to `<value>null</value>` and removing the other "unset" related code (including the entry in the database)?

I'm just trying to understand your proposal better.  It seems like we *do need* some sort of `<value>` tag which represents the idea that this External Service supports "regular Items" as well.  So, that implies we need to either have that represented by something like the current "unset" implementation, or `<value>null</value>` or maybe something like either `<value>item</value>` or `<value></value>`.  

Overall, I agree with you that I'd rather see if we can find a way to not require an "unset" Entity Type, but I'm not sure I understand your idea for how to do so.

Reply:
Use `<null/>` in place of `<value>something</value>`.
https://mkyong.com/spring/how-to-inject-null-value-in-spring/

Reply:
Regardless, if we are to have a way to express this idea, then non-entity Item must have a type-name, whether it is "unset", "item", `null`, "", or "Aloysius".  It means "Item with no value for dspace.entity.type", and is in effect a type of entity having zero relationships.  If this type is not defined with the other entity types, then it must be a special case in the code.  The special-case code may come to very little:  `null` won't match any LeftType or RightType.

Reply:
`null` don't work well in our REST architecture. There is no way to expose a `null` entity type reference. It is similar to the `anonymous` group that exists in the database just to make the life easier and have a more uniform code base

Reply:
@abollini & @benbosman : As discussed in today's meeting, I think the current approach is "good enough" as it helps us resolve a [major usability issue that makes ORCID Entity integration almost unusable](https://github.com/DSpace/dspace-angular/issues/1095).  However, I suggest renaming `unset` to be `none`.  

A few additional thoughts out of today's discussion:
* We should add comments/documentation to clarify that an Entity Type of `none` means that the `dspace.entity.type` metadata field does not exist on the Item.  So, no Items should ever have a `dspace.entity.type=none` (but we don't have a way to really "enforce" that rule at this time)
* We will need to add documentation to warn everyone to never use/rename/delete this `none` type as it's required to be in the database table.  
* Because `none` is required in the database table, it also means that the `none` type MUST be added to the database via a Flyway Callback (similar to how the Anonymous/Administrator groups are added in [GroupServiceInitializer](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/storage/rdbms/GroupServiceInitializer.java)).  The reason a Flyway Callback must be used is so that the `none` type can be auto-restored (if it's ever deleted or renamed accidentally) the next time you run a database migration.  _(NOTE: This will require changes to this PR, as I see this PR currently just creates a new migration to add this value)_
* We need to minimally ensure the `none` type cannot be used from the User Interface (e.g. selected for a Collection, etc).  Either we need to work around it in UI code, or the REST API could return the `none` type with an immutable status flag that lets the UI know it cannot be modified.
* As Ben noted, since the UI is not the only place using Entities, ideally any default *Entities configuration* which references the `none` type should include a comment to describe it.  So, the `external-services.xml` config above might say something like `<!-- Allow PubMed import to also be used with regular Items (which have no entity type) -->`   These comments can better document the reason for any usages of `none` to help discourage anyone for using it in other ways.

If I missed anything, or if anything needs more discussion, let me know.